src/Libraries/Nop.Services/Helpers/BrowscapXmlParser.cs

```diff
@@ -14,12 +15,11 @@ namespace Nop.Services.Helpers
     /// </summary>
     public class BrowscapXmlHelper
     {
-        private readonly List<string> _crawlerUserAgentsRegexp;
+
+        private ConcurrentDictionary<String, bool> _userAgentsCache;
```

Comment:
I don't think using ConcurrentDictionary is necessary for this task because this list a never changes after creating .